pytest-embedded-idf/pytest_embedded_idf/dut.py

```diff
@@ -351,12 +167,20 @@ def _parse_unity_menu_from_str(s: str) -> t.List[UnittestMenuCase]:
 
         return test_menu
 
+    def _hard_reset(self):
+        if hasattr(self, 'serial'):
+            return self.serial.hard_reset()
+        elif hasattr(self, 'qemu'):
+            return self.qemu._hard_reset()  # noqa
```

Comment:
This looks a bit hacky, since this prevents new DUT classes being added without modifying this file. Plus, we can forget to modify this file when a new DUT is added...

Can we perhaps let the DUT register a callback in the mixin (`self._hard_reset_func`)?

Reply:
sure. applied with your suggestions